src/phylo/iteration.jl

```diff
@@ -0,0 +1,277 @@
+#==================================#
+# Iteration and traversal of trees #
+#==================================#
+
+# Ben J. Ward, 2014.
+
+@Docile.doc """
+`PhylogenyIterator` is an abstract type that defines a family of iterators
+for traversing trees in various ways, including `BreadthFirst`, and `DepthFirst`,
+or from a tip to a root.
+""" ->
+abstract PhylogenyIterator
```

Comment:
I don't know that this abstract type is needed. Abstract types are useful to groups types that share some common interface, but there's not much of a specialized interface here. These are all just regular iterators.